### DISCUSSION TITLE
Functionalities of `HardLimiter` and `Limiter`

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#173] [Modeling] [answered]
@cuihantao 
Hi Hantao,

I noticed that the doc of `HardLimiter` said it is an alias of `Limiter`, and I'm curious that whether the `HardLimiter` can actually limit a variable in ANDES? Since `Limiter` cannot do that, I don't think `HardLimiter` can do that?

This will affect the `PIAWHardLimit` since the output is added with a `Limiter`.  Does it mean the output may not be actually limited?
If this is true maybe we should replace the output `Limiter` with an `AntiWindup`?

Regards,
Jinning

**Comment by cuihantao** [comment, marked as answer]
`HardLimiter` and `Limiter` do not actually limit the variables. They process the input variables and spits out the flags, `zi`, `zl`, and `zu`. It is up to the user to construct the equation, making use of the flags. 

Use `GainLimiter` if you want the outputs to be pegged at limits when hit.

For `PIAWHardLimit`, the output is its `y` variable, which already uses the limiter output flags.

**Reply by jinningwang** [reply]
Thanks for your reply.

Then I have another question: will `AntiWindup` realize the function of 'actual limit variables'? If so we don't need to build the equations with Limiter flags.

Regards,
Jinning

**Reply by cuihantao** [reply]
Yes, `AntiWindup` is a special case where the variables are set in `TDS`. It can not be set through its own equation because it depends on the equation.

Search for `system.antiwindup` in `TDS.py`.

**Reply by jinningwang** [reply]
Thanks, I'll take a look at it.

Regards,
Jinning

**Comment by cuihantao** [comment]
Docs improved. Thanks! @jinningwang
